### ISSUE TITLE
Image.Name is null when CloudServersProvider.ListServersWithDetails method is called

### ISSUE THREAD

**Issue by DonSchenck** [openstacknetsdk/openstack.net#339]
Image is a child object of the Server object returned from the method call.

The .Name property of the Image (e.g. Server.Image.Name) is null.

**Comment by carolynvs** [comment]
Digging into this further, and I see why it's empty. The details endpoint only returns the Image id. So if you need the name that will be a second request.

I am revamping things a bit though so that this isn't so hard:

```csharp
var compute = new ComputeService(...);
var servers = compute.ListServerDetails();
foreach(var server in servers)
{
  ImageReference imageRef = server.Image;
  Image image = imageRef.GetImage();
  Console.WriteLine(image.Name);
}
```

**Comment by carolynvs** [comment]
It's awkward when sometimes an endpoint will return the name and other times it doesn't. In this case, when you hit /servers/server-id, it returns a reference to an image, with only the id populated. When you hit /images/, then it lists references to images with both id and name populated. The same type of pattern (where sometimes a reference has just an id, or just a name), occurs elsewhere like security groups.

At the moment in 1.6.0-beta1, I am using the same class for both cases, and documenting on the Name property that it may not always be populated. The code sample above demonstrates how to grab the full image definition to get at the name.

I don't like that it's not always clear when the user can expect to have a field populated. I'm not sure if it would help or hurt to use different classes in the two cases above, or what to call each? Right now when one object refers to another, I'm calling the class `FooReference`. But maybe I need 3 class names:

* `Foo` - the full resource
* `FooReference` - when a resource references another by name or id
* `Foo???` - when the list endpoint for a resource doesn't return full resources.

@DonSchenck Any suggestions for that last one? Maybe `FooLink`?

**Comment by carolynvs** [comment]
I have a name for `Foo???`... `FooSummary`. :smile:

**Comment by carolynvs** [comment]
Fixed by #590
